apps/linux/vim.talon

```diff
@@ -0,0 +1,799 @@
+    key(ctrl-w)
+    key(p)
+split top left:
+    # XXX - until we have key combo support
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(t)
+split next:
+    # XXX - until we have key combo support
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(w)
+split (previous|prev):
+    # XXX - until we have key combo support
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(W)
+split bottom right:
+    # XXX - until we have key combo support
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(b)
+split preview:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(P)
+
+# personal convenience shortcuts
+# split right
+sprite:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(l)
+
+# split left
+spleff:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(h)
+
+# split top left
+splot:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(t)
+
+sprot:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(b)
+
+
+# moving windows
+split (only|exclusive):
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(o)
+split rotate [right]:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(r)
+split rotate left:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(R)
+split move top:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(K)
+split move bottom:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(J)
+split move right:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(L)
+split move left:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(H)
+split (move to|make) tab:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(T)
+
+# window resizing
+split (balance|equalize):
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(=)
+
+# atm comboing these with ordinals is best, but may add number support
+split taller:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(+)
+    user.vim_set_normal_mode_exterm()
+split shorter:
+    user.vim_set_normal_mode_exterm()
+    key(ctrl-w)
+    key(-)
+    user.vim_set_normal_mode_exterm()
+split fatter:
+    key(ctrl-w)
+    key(>)
+    user.vim_set_normal_mode_exterm()
+split skinnier:
+    key(ctrl-w)
+    key(<)
+set split width:
+    user.vim_normal_mode_exterm(":resize ")
+set split height:
+    user.vim_set_normal_mode_exterm(":vertical resize ")
+    insert(":vertical resize ")
+
+###
+# Diffing
+###
+(split|window) start diff:
+    user.vim_set_normal_mode_exterm(":windo diffthis\n")
+
+(split|window) end diff:
+    user.vim_set_normal_mode_exterm(":windo diffoff\n")
+
+buffer start diff:
+    user.vim_set_normal_mode_exterm(":bufdo diffthis\n")
+
+buffer end diff:
+    user.vim_set_normal_mode_exterm(":bufdo diffthis\n")
+
+###
+# Tab
+###
+(list|show) tabs: user.vim_normal_mode(":tabs\n")
+(close this tab|tab close): user.vim_normal_mode_exterm(":tabclose\n")
```

Comment:
I would suggest implementing the existing tab-related actions in `app` where possible, so the common tab grammar can be used as much as possible 

https://github.com/knausj85/knausj_talon/blob/master/misc/tabs.talon

Reply:
I hadn't noticed those existed, will do

Reply:
I was thinking about this while I was going to add it, and there is an inconsistency in the repo with regards to how generic command order is approached across the repo, specifically with the order of [subject][action] vs [action][subject]. The most obvious example is earlier in this review you recommended using the `file save`, which follows the `[subject][action]` order. However, with the generic tab commands they all support the `[action][subject]` order. I also mix up the order situationally within the vim grammars, but it's making me wonder if we should try to standardize on one over the other, since I know you don't want to have duplicate ways of saying commands in your own repo.

I personally prefer [subject][action] in general because I can speak the subject that I want to interact with and then say what I want to do with it and it helps me think about what commands I actually going to give. I also think it correlates well to command-line stuff, where we say the number of the command first, and then the sub command. That said I guess it will rely depend on the user as to what order seems most intuitive though. I can move this discussion into a separate issue if you prefer.

Reply:
I hadn't really consciously considered this, but it's a great point. 

The tab commends were based on how it's spoken in English... "open a new tab and go here..." whereas file save was probably based on the the menu order.

That said, I think this should be a separate issue -
https://github.com/knausj85/knausj_talon/issues/148